online-inference/stable-diffusion/03-inference-service.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: serving.kubeflow.org/v1beta1
+kind: InferenceService
+metadata:
+  name: stable-diffusion
+spec:
+  predictor:
+    containerConcurrency: 1
+    minReplicas: 1
+    maxReplicas: 1 
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: gpu.nvidia.com/class
+              operator: In
+              values:
+              - RTX_A4000
+            - key: topology.kubernetes.io/region
+              operator: In
+              values:
+              - ORD1 
+    containers:
+      - name: kserve-container
+        image: tweldoncw/stable-diffusion:4
+        command:
+          - "python3"
+          - "/app/service.py"
+          - "--model-id=CompVis/stable-diffusion-v1-4"
+          - "--hf-home=/mnt/models/hub"
+          # The following arguments are defaults which can be changed here and per request
+          - "--precision=float16" # PyTorch dtype/model precision(one of float16 or float32)
+          - "--guidance-scale=7.0" # How well image generation adheres to the prompt. With very large numbers the images might look good, but will be less diverse. 
+          - "--num-inference-steps=50" # Results are better the more steps you use but will take longer.
+          - "--seed=42" # Every time you use the same seed you'll have the same image result. Use random numbers for different results with the same prompt.
```

Comment:
Are these defaults or what's the purpose of these specified here?

Reply:
Feels redundant since defaults are also specified in the API Code further down

Reply:
If users want to change the defaults(and yes these values match the code defaults) they can see all options available and change them if desired in one place.

For other models we've included the defaults in the manifest as ENV vars, this is just a stylistic change to CLI arguments. 

Can change back to ENV vars, or remove as yes it is redundant(but IMO convenient).

Reply:
But defaults are here too right: https://github.com/coreweave/kubernetes-cloud/pull/88/files#diff-bf2cb0ebe5fe364e694c1d62e31788f44aebcb7db01d531fa306074e66e97346R17 feels duplicated?

Reply:
yes it is duplicated. I personally like being able to see the options available from the manifest while I'm working on these because I'd prefer to change values from there when experimenting with different values. 

Some values can be changed per request, some cannot. I've rearranged args to properly indicate what can be changed and when. 

I can either remove these because they are admittedly redundant, or perhaps comment them out so the code is acting as the default but users can easily see what options are available just from the manifest if they want to change anything?

Reply:
Removed